sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -2230,6 +2230,8 @@ class Analyzer(
           val result = resolved transformDown {
             case UnresolvedMapObjects(func, inputData, cls) if inputData.resolved =>
               inputData.dataType match {
+                case ArrayType(et, false) if cls.isEmpty =>
```

Comment:
is it really safe to do so? The `MapObject` is not only used for null checking, but also to resolve struct in array.